datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1487,6 +1489,17 @@ fn build_predicate_expression(
                 *bin_expr.op(),
                 bin_expr.right().clone(),
             )
+        } else if let Some(like_expr) = expr_any.downcast_ref::<phys_expr::LikeExpr>() {
+            if like_expr.case_insensitive() {
+                return unhandled_hook.handle(expr);
+            }
+            let op = match (like_expr.negated(), like_expr.case_insensitive()) {
+                (false, false) => Operator::LikeMatch,
+                (true, false) => Operator::NotLikeMatch,
+                (false, true) => Operator::ILikeMatch,
+                (true, true) => Operator::NotILikeMatch,
```

Comment:
I think this is dead code as `if like_expr.case_insensitive() {` catches the case insensitive case

I think this code would be clearer if it just matched on `like_expr.negated()` (or alternately returned `unhandled_hook.handle(expr);` directly for these last 2 cases

Reply:
Sure will change it to do so. I think I was getting a bit ahead of myself to implement `ILIKE` support, which as per the comment should be possible, maybe you can show me how to construct the physical expression to call `lower()` and `upper()` on another expression.